src/mlpack/methods/ann/ffn.hpp

```diff
@@ -172,9 +172,10 @@ class FFN
    * @param i Index of points to use for objective function gradient evaluation.
    * @param gradient Matrix to output gradient into.
    */
+  template<typename eT>
   void Gradient(const arma::mat& parameters,
                 const size_t i,
-                arma::mat& gradient);
+                arma::Mat<eT>& gradient);
```

Comment:
Ahh.. ooofff. Took me a while to figure out why I actually did this :) The change successfully enables the use of FNN as a controller in NTM. So with this w don't have to create a new controller layer type. A user can create an FFN, like he always does, and pass it to NTM. NTM will use this as the controller. Although I do have a question here. Here the user can add recurrent layers to this FFN cause the underlying call is made by RNN of NTM. Thats why I did not make the changes to RNN. Do you think thats confusing? Should I also make RNN LayerType supported?